servicehandler/servicediscovery.go

```diff
@@ -0,0 +1,203 @@
+package servicehandler
+
+import (
+	"context"
+	"slices"
+	"sync"
+	"time"
+
+	logger "github.com/kubescape/go-logger"
+	"github.com/kubescape/k8s-interface/k8sinterface"
+	"github.com/panjf2000/ants/v2"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	k8sruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	kind         = "ServiceScanResult"
+	resource     = "servicesscanresults"
+	group        = "kubescape.io"
+	version      = "v1"
+	apiVersion   = group + "/" + version
+	fieldManager = "kubescape|operator|serviceDiscoveryHandler"
+	workerNum    = int(20)
+)
+
+var protocolFilter = []string{"UDP"}
+
+var serviceListOptions = metav1.ListOptions{
+	FieldSelector: "metadata.namespace!=kube-system",
+}
+
+var ServiceScanSchema = schema.GroupVersionResource{
+	Group:    group,
+	Version:  version,
+	Resource: resource,
+}
+
+func deleteServices(ctx context.Context, client dynamic.NamespaceableResourceInterface, currentServicesMetadata []metadata) {
+	// get all services from the current cycle and compare them with the current CRDs
+
+	authServices, err := client.List(ctx, metav1.ListOptions{})
+	if err != nil {
+		logger.L().Ctx(ctx).Error(err.Error())
+		return
+	}
+
+	for _, service := range authServices.Items {
+		crdMetadata := metadata{
+			name:      service.GetName(),
+			namespace: service.GetNamespace(),
+		}
+
+		if !slices.Contains(currentServicesMetadata, crdMetadata) {
+			err := client.Namespace(service.GetNamespace()).Delete(ctx, service.GetName(), metav1.DeleteOptions{})
+			if err != nil {
+				logger.L().Ctx(ctx).Error(err.Error())
```

Comment:
if there's an error, continue to next item  or return error (decide based on the logic)